api/http/handler/registry.go

```diff
@@ -318,3 +353,243 @@ func (handler *RegistryHandler) handleDeleteRegistry(w http.ResponseWriter, r *h
 		return
 	}
 }
+
+// proxyRequestsToRegistryAPI proxy to registry and set auth header if necessary
+func (handler *RegistryHandler) proxyRequestsToRegistryAPI(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	id := vars["id"]
+
+	registryID, err := strconv.Atoi(id)
+	if err != nil {
+		httperror.WriteErrorResponse(w, err, http.StatusBadRequest, handler.Logger)
+		return
+	}
+
+	registry, err := handler.RegistryService.Registry(portainer.RegistryID(registryID))
+	if err == portainer.ErrRegistryNotFound {
+		httperror.WriteErrorResponse(w, err, http.StatusNotFound, handler.Logger)
+		return
+	} else if err != nil {
+		httperror.WriteErrorResponse(w, err, http.StatusInternalServerError, handler.Logger)
+		return
+	}
+
+	client := &http.Client{
+		Timeout: registryTimeout,
+	}
+	if !registry.TLSVerification {
+		client.Transport = &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		}
+	}
+
+	// Build first request (without auth)
+	uri := strings.Replace(r.RequestURI, "/api/registries/"+id+"/", "", 1)
+	req := r
+	req.RequestURI = ""
+	u, err := url.Parse(registry.Protocol + "://" + registry.URL + "/" + uri)
```

Comment:
It's for transfomration between the server request (r) and the new client request (req)
Regarding doc, they have difference for URL and RequestURI fields
https://golang.org/pkg/net/http/#Request

Reply:
All right but what about simply copying the request like this:

```golang
	req, err := http.NewRequest(request.Method, registry.Protocol + "://" + registry.URL, nil)
	if err != nil {
		return nil, err
	}
```

Looks a lot simpler to me. Some additions might be required if we need to copy the headers, do we?

Reply:
Yeah that's an other solution, but i think we are more generally if we copy all and modify just the necessary. It allow us to add a header or whatever to original request (from the cli) and it will be copy to the target registry (like a proxy)
I'll add some comments and refactor line order to be more clear on what code do, tell me what you thinkk

Reply:
My opinion is that if you're solution do not bring something particular (besides being "more generally" which I'm not sure is useful in our case), I believe that what I proposed is easier to read. I'd go for my proposition.